windows/win_zip.ps1

```diff
@@ -0,0 +1,161 @@
+#!powershell
+# This file is part of Ansible
+#
+# Copyright 2015, David O'Brien <obrien.david@outlook.com>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+Function Test-Key
+{
+  param (
+    [string]$path,
+    [string]$key
+  )
+
+  if (!(Test-Path $path)) {
+    return $false
+  }
+  if ((Get-ItemProperty $path).$key -eq $null) {
+    return $false
+  }
+  return $true
+}
+
+Function Get-FrameworkVersions
+{
+  $installedFrameworks = @()
+  if (Test-Key "HKLM:\Software\Microsoft\NET Framework Setup\NDP\v4\Client" "Install") {
+    $installedFrameworks += "4.0c"
+    If ((Get-ItemProperty "HKLM:\Software\Microsoft\NET Framework Setup\NDP\v4\Client").Version -like "4.5*") {
+      $installedFrameworks += "4.5c"
+    }
+  }
+  If (Test-Key "HKLM:\Software\Microsoft\NET Framework Setup\NDP\v4\Full" "Install") {
+    [int32]$intRelease = (Get-ItemProperty "HKLM:\Software\Microsoft\NET Framework Setup\NDP\v4\Full").Release
+    Switch ($intRelease)
+    {
+      "378389" { $installedFrameworks += "4.5" }
+      "378675" { $installedFrameworks += "4.5.1" }
+      "378758" { $installedFrameworks += "4.5.1" }
+    }
+  }
+  return $installedFrameworks
+}
+
+$params = Parse-Args $args;
+
+$result = New-Object psobject @{
+  win_zip = New-Object psobject
+  changed = $false
+}
+
+# Check if all arguments were provided and set defaults for optional params
+If ($params.src) {
```

Comment:
I don't have a 2.0 dev environment at the moment. So I have to postpone this for now.